src/libraries/Common/src/System/Net/Http/Http3/Helpers/VariableLengthIntegerHelper.cs

```diff
@@ -0,0 +1,213 @@
+                            bytesRead = 4;
+                            return true;
+                        }
+                        break;
+                    default: // LengthEightByte
+                        Debug.Assert((firstByte & LengthMask) == LengthEightByte);
+                        if (BinaryPrimitives.TryReadUInt64BigEndian(buffer, out ulong serializedLong))
+                        {
+                            value = (long)(serializedLong - EightByteSubtract);
+                            Debug.Assert(value >= 0 && value <= EightByteLimit, "Serialized values are within [0, 2^62).");
+
+                            bytesRead = 8;
+                            return true;
+                        }
+                        break;
+                }
+            }
+
+            value = 0;
+            bytesRead = 0;
+            return false;
+        }
+
+        public static bool TryRead(ref SequenceReader<byte> reader, out long value)
+        {
+            // Hot path: we probably have the entire integer in one unbroken span.
+            if (TryRead(reader.UnreadSpan, out value, out int bytesRead))
+            {
+                reader.Advance(bytesRead);
+                return true;
+            }
+
+            // Cold path: copy to a temporary buffer before calling span-based read.
+            return TryReadSlow(ref reader, out value);
+
+            static bool TryReadSlow(ref SequenceReader<byte> reader, out long value)
+            {
+                ReadOnlySpan<byte> span = reader.CurrentSpan;
+
+                if (reader.TryPeek(out byte firstByte))
+                {
+                    int length =
+                        (firstByte & LengthMask) switch
+                        {
+                            LengthOneByte => 1,
+                            LengthTwoByte => 2,
+                            LengthFourByte => 4,
+                            _ => 8 // LengthEightByte
+                    };
+
+                    Span<byte> temp = (stackalloc byte[8])[..length];
+                    if (reader.TryCopyTo(temp))
+                    {
+                        bool result = TryRead(temp, out value, out int bytesRead);
+                        Debug.Assert(result == true);
+                        Debug.Assert(bytesRead == length);
+
+                        reader.Advance(bytesRead);
+                        return true;
+                    }
+                }
+
+                value = 0;
+                return false;
+            }
+        }
+
+        public static long GetInteger(in ReadOnlySequence<byte> buffer, out SequencePosition consumed, out SequencePosition examined)
+        {
+            var reader = new SequenceReader<byte>(buffer);
+            if (TryRead(ref reader, out long value))
+            {
+                consumed = examined = buffer.GetPosition(reader.Consumed);
+                return (long)value;
```

Comment:
`value` is already declared as `long`.  Why are we casting to `long`?

Reply:
Leftover from me trying to use proper unsigned types. Will remove.